upi/openstack/network.yaml

```diff
@@ -5,40 +5,146 @@
 # openstacksdk
 # netaddr
 
+- import_playbook: common.yaml
+
 - hosts: all
   gather_facts: no
 
   tasks:
   - name: 'Create the cluster network'
     openstack.cloud.network:
       name: "{{ os_network }}"
+    when: os_networking_type != "CiscoACI"
+
+  - name: 'list network'
+    command:
+      cmd: "openstack network list --name {{ os_network }} --tag {{ cluster_id_tag }}"
+    when: os_networking_type == "CiscoACI"
+    register: cluster_network
 
-  - name: 'Create the cluster IPv4 subnet'
-    openstack.cloud.subnet:
+  - name: 'get network epg contract masters attributes'
+    command:
+      cmd: "neutron net-show {{ os_network }} -f json -c apic:epg_contract_masters"
+    when:
+      - os_networking_type == "CiscoACI"
+      - cluster_network.stdout != ""
+    register: epg_bd
+
+  - name: 'convert network show output into dictionary type'
+    set_fact:
+         epg_bd_out: "{{ epg_bd.stdout| from_json }}"
+    when:
+      - os_networking_type == "CiscoACI"
+      - cluster_network.stdout != ""
+      - epg_bd.stdout != ""
+
+  - name: 'set facts for network epg contract masters attributes'
+    set_fact:
+         app_profile_name: "{{ epg_bd_out['apic:epg_contract_masters'].0.app_profile_name }}"
+         node_epg: "{{ epg_bd_out['apic:epg_contract_masters'].0.name }}"
+    when:
+      - os_networking_type == "CiscoACI"
+      - cluster_network.stdout != ""
+      - epg_bd_out is defined
+
+  - name: 'Create the cluster network with aci-containers-nodes EPG contract relationship for node network'
+    command:
+      cmd: "neutron net-create {{ os_network }} --apic:epg_contract_masters list=true type=dict app_profile_name={{ aci_cni['app_profile'] }},name={{ aci_cni['node_epg'] }} --apic:distinguished_names type=dict BridgeDomain={{ aci_cni['network_interfaces']['node']['bd'] }}"
+    when:
+      - os_networking_type == "CiscoACI"
+      - (cluster_network.stdout == "" ) or
+        (app_profile_name is defined and app_profile_name != aci_cni['app_profile']) or
+        (node_epg is defined and node_epg != aci_cni['node_epg'])
+
+  - name: 'Create a subnet'
+    os_subnet:
       name: "{{ os_subnet }}"
       network_name: "{{ os_network }}"
       cidr: "{{ os_subnet_range }}"
       allocation_pool_start: "{{ os_subnet_range | ansible.utils.next_nth_usable(10) }}"
       allocation_pool_end: "{{ os_subnet_range | ansible.utils.ipaddr('last_usable') }}"
       dns_nameservers: "{{ os_external_dns }}"
+    when: os_networking_type != "CiscoACI"
 
-  - name: 'Create the cluster IPv6 subnet'
```

Comment:
@PariyaAshok why we are removing upstream changes? Our changes should come on top of it. It will not be processed if we not define os_subnet6, but we should not remove it

Reply:
thanks @abhis2112, good catch, it was removed as part of meld, I added back.